Applied filtering logic to goal buddy cards

KevinMeisenbacher opened the pull request:
Refactored and cleaned up states to make my filtering system dynamically repopulate the goal buddy cards accordingly
✅Replaced all instances of GoalBuddyList in src/pages/ColabPage.tsx with goalBuddiesMergedWithUsers
✅Changed type for filteredGoalBuddies state to match goalBuddiesMergedWithUsers
✅Moved the function for rendering goalBuddy cards upstairs to display all goal buddies if all filters are off and display filtered goal buddies only if filters are on
![image](https://github.com/user-attachments/assets/74f968ac-4a35-49d9-8baa-8229539a3836)
![image](https://github.com/user-attachments/assets/31952d46-59c1-43eb-bb4d-8524889f551a)
![image](https://github.com/user-attachments/assets/675e017b-26d0-4ce6-91c6-57131ad8e33b)
Keep in mind that I use dark mode because bright LEDs hurt my eyes. When you test my code, your colours will match the prototype.

f2xiao commented:
applied filter, does change te data, but how do I test it's filtering the right data?

KevinMeisenbacher commented:
The filtering is all frontend. Also took all logs out. Do you see any changes in how the cards populate? That's all you should see. This commit doesn't make any API requests because that would be a waste of data, code and work. Only thing that matters is what the user sees.

KevinMeisenbacher commented:
Just pulled and merged develop then fixed merge conflicts. You guys were right about the filter not working. It seemed to when I pushed up, so I guess it broke. Fixed it after finding the issue, so give it another look. Note that the styling update will be included in ticket 114, but everything should filter properly now, and the skeleton should show.

JeffreyChuCPA commented:
Works as aspected but a few things to change
1. you should updated the type for filterGoalBuddies in the filterProps to be the type of the filterGoalbuddies function you defined ((choice: string) => void)
2. There is no merge conflicts, but I had to deal with merge conflicts when I merged my skeleton PR (or some other PR, cant rmb exactly). So if you look at the current ColabPage in develop, you should adapt your ColabPage in this PR to be functionally the same. Essentially the change I remember doing was moving the isLoading logic and the subsequent render logic to be nested and after your Filter component. In your current Colabpage, the isLoading is before section and FIlter. To ensure theres not a bug after you merge, you should change your Colabpage to be similar to whats in the dev branch. You can visually tell your colab page is not up to date with the change I made to the ColabPage in regards to the skeleton. See pics for reference: 
![image](https://github.com/user-attachments/assets/65696d91-23c1-4de0-85cb-87053992f692)
![image](https://github.com/user-attachments/assets/5fe53af0-9c1a-4565-8bcc-cebc5308dbe6)

JeffreyChuCPA commented on src/pages/ColabPage/ColabPage.tsx:
```diff
@@ -77,38 +72,49 @@ export default function ColabPage() {
     applyFilter()
   }, [filter])
 
+  const renderGoalBuddies = (goalBuddies: any[]) => {
```
change the any type to be AllGoalBuddyData[] since this should render either filtered or full lists that are of that type

KevinMeisenbacher commented:
![image](https://github.com/user-attachments/assets/d1001601-5cf6-400e-bcc7-e533bc561437)
I already tried that, but it doesn't work which is why I made it any because I figure that if it looks good and works well, then it's good. Is any strictly forbidden in a professional setting?

KevinMeisenbacher commented:
> Works as aspected but a few things to change
> 
> 1. you should updated the type for filterGoalBuddies in the filterProps to be the type of the filterGoalbuddies function you defined ((choice: string) => void)
> 2. There is no merge conflicts, but I had to deal with merge conflicts when I merged my skeleton PR (or some other PR, cant rmb exactly). So if you look at the current ColabPage in develop, you should adapt your ColabPage in this PR to be functionally the same. Essentially the change I remember doing was moving the isLoading logic and the subsequent render logic to be nested and after your Filter component. In your current Colabpage, the isLoading is before section and FIlter. To ensure theres not a bug after you merge, you should change your Colabpage to be similar to whats in the dev branch. You can visually tell your colab page is not up to date with the change I made to the ColabPage in regards to the skeleton. See pics for reference:
>    ![image](https://private-user-images.githubusercontent.com/141854177/413657453-65696d91-23c1-4de0-85cb-87053992f692.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MDkzMTMsIm5iZiI6MTczOTgwOTAxMywicGF0aCI6Ii8xNDE4NTQxNzcvNDEzNjU3NDUzLTY1Njk2ZDkxLTIzYzEtNGRlMC04NWNiLTg3MDUzOTkyZjY5Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxN1QxNjE2NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wZjhlMzNmMDNhYWRkN2Q3YjYyMWUwN2EwN2M5ZWU0NDg5NDNlZGMxOGEwN2Y1YjhjODdmODMyMDcwMWNiNjU2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.0ljvDQmJwBsijA4aKCbUTZv0GnBNORUOnPO-LIsy6fQ)
>    ![image](https://private-user-images.githubusercontent.com/141854177/413657474-5fe53af0-9c1a-4565-8bcc-cebc5308dbe6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MDkzMTMsIm5iZiI6MTczOTgwOTAxMywicGF0aCI6Ii8xNDE4NTQxNzcvNDEzNjU3NDc0LTVmZTUzYWYwLTljMWEtNDU2NS04YmNjLWNlYmM1MzA4ZGJlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxN1QxNjE2NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03MTY1MmJjYjcyZWFlYmRmYmRiMjAzYjdmNzA1YzZiZGJhNWVhZDE5OTg1ZmFmMTIwOTkyNWRjZmE0OGU3M2IwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HAk3m0CEGN6mEGnEnPGnyM1CdhM4uZNaywj61NVgigQ)
>    ![image](https://private-user-images.githubusercontent.com/141854177/413657545-c6f94698-f34b-42fe-9ede-746f389f1cbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MDkzMTMsIm5iZiI6MTczOTgwOTAxMywicGF0aCI6Ii8xNDE4NTQxNzcvNDEzNjU3NTQ1LWM2Zjk0Njk4LWYzNGItNDJmZS05ZWRlLTc0NmYzODlmMWNiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIxN1QxNjE2NTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZDdmMzE3MTNiNTNjY2I2ZmVjMThkYWM5YzRlMDQ2MmE0M2Y2Yjg0MTg1MTAwNGUwNWE5Zjc2ZDljZTVlMGY1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2dEEyOJzGExqNeung_1rsQoU0ppJwTCqQ0NzkGo67Ko)

Thanks for the feedback. I tried that out, but it looks like all of the data is gone, so I'll try again when it's back.

KevinMeisenbacher commented:
I put the filter inside of the loading section because it doesn't make sense for it to be there while the page is loading, but regardless, as per your requests, I moved it outside. I'm keeping renderGoalBuddies as any because changing it to the right type causes incompatibilities that would take too much time to figure out how to fix, so I'll leave that for the final week because what matters now is merging our stuff together so it looks good and works well before the sun sets.

KevinMeisenbacher commented:
Just refactored, fixed merged conflicts, tested, awaited db restoration, fixed, pulled, merged, fixed merge conflicts, refactored and fixed, pushed, committed, pulled, fixed merged conflicts, merged, fixed merge conflicts, committed and pushed back up with necessary recommendations addressed. HOOOOOOOOOOH boy! 

I had the filter in the loading section because I think it looks better for it to show up only when the page is loaded, but I moved it outside as per requested because if that reduces merge conflicts, then I'm totally down for that! I tried setting the type of RenderGoalBuddies to AllGoalBuddyData, but that caused incompatibility issues that would take too much time to fix, so I'm keeping that as-is until final week because it's more important to get code looking good and working well before the sun sets. Please give my code another look and let me know whether it's good to merge.

KevinMeisenbacher commented:
> Can confirm that skeleton looks as intended. Filter looks to be working (without test cases to specifically check going off of matching combinations to what appears in individual filters and making sure all that appear in each individual show up in combined). Did not see any errors or console logs in the the console.

The middle part of your comment definitely has a lot of words. Does that mean there's something I need to fix, or am I good to merge?

JeffreyChuCPA commented:
Doing another quick check rn, after that I can say you are good to merge. This would be a lot easier to test when my PR about the styling goal buddy card is merged after, but that will be after the fact to see if the filter is working 100% fine visually.

JeffreyChuCPA commented:
> ![image](https://private-user-images.githubusercontent.com/55214897/413923229-d1001601-5cf6-400e-bcc7-e533bc561437.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MjI0MjAsIm5iZiI6MTczOTgyMjEyMCwicGF0aCI6Ii81NTIxNDg5Ny80MTM5MjMyMjktZDEwMDE2MDEtNWNmNi00MDBlLWJjYzctZTUzM2JjNTYxNDM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMjE3VDE5NTUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4MjkxYTM3ZDgwNDY1ZjQ5OGFiNGVkYTM5YmYzYTczNWRlMGZjYTg5MDM2NjYwNGUwMjk1NTI0YzE2YzNhZjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5o03mz4tlhlvdRZrc8F5bVALJGF5kiTmXwsqijjoKVs) I already tried that, but it doesn't work which is why I made it any because I figure that if it looks good and works well, then it's good. Is any strictly forbidden in a professional setting?

I would think having any is fine if theres a specifc/good reason for it. For this, it makes sense to explicitly state the types that this function takes and what it would return. You can make the following change below in replacement of the any []:

```
  const renderGoalBuddies = (goalBuddies: AllGoalBuddyData[]): ReactNode => {
    return goalBuddies.map((goalBuddyWithUser) => (
      <GoalBuddyCard key={goalBuddyWithUser.id} goalBuddy={goalBuddyWithUser} />
    ))
  }
```

JeffreyChuCPA commented on src/pages/ColabPage/ColabPage.tsx:
```diff
@@ -77,38 +72,43 @@ export default function ColabPage() {
     applyFilter()
   }, [filter])
 
+  const renderGoalBuddies = (goalBuddies: any[]) => {
+    goalBuddies = goalBuddies.map((goalBuddyWithUser) => (
+      <GoalBuddyCard
+        key={goalBuddyWithUser.id}
+        goalBuddy={goalBuddyWithUser}
+      />
+    ))
+    return goalBuddies
+  }
+
```
See previous comment about how to address the any[] here

KevinMeisenbacher replied:
Fixed. Good to merge, then?

Redi-Molla commented:
Fix the merge conflict. Also, you said you addressed Jeffery's comment, but I don't see the change. Did you push it to the remote branch?

KevinMeisenbacher commented:
Yeah. We might have to postpone the push to main, then. I'll do a Google
meet at 7 to resolve this.

On Mon, Feb 17, 2025, 4:29 p.m. Rediet Molla ***@***.***>
wrote:

> Fix the merge conflict. Also, you said you addressed Jeffery's comment,
> but I don't see the change. Did you push it to the remote branch?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/makeitMVPadmin/LAP6_colab/pull/33#issuecomment-2664103489>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANFIGMNT2JLMMC66R4JDJS32QJICJAVCNFSM6AAAAABXFNV3TSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRUGEYDGNBYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: Redi-Molla]*Redi-Molla* left a comment
> (makeitMVPadmin/LAP6_colab#33)
> <https://github.com/makeitMVPadmin/LAP6_colab/pull/33#issuecomment-2664103489>
>
> Fix the merge conflict. Also, you said you addressed Jeffery's comment,
> but I don't see the change. Did you push it to the remote branch?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/makeitMVPadmin/LAP6_colab/pull/33#issuecomment-2664103489>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ANFIGMNT2JLMMC66R4JDJS32QJICJAVCNFSM6AAAAABXFNV3TSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDMNRUGEYDGNBYHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
